### DISCUSSION TITLE
Recommend using "Access = Internal" on codeunits which does not expose any external functionality (e.g. procedure, event) (Severity = Info)

### DISCUSSION THREAD

**Discussion by someC0d3r** [StefanMaron/BusinessCentral.LinterCop#810] [New Rule]
See title

I don't know if any other object type make sense, but for codeunits im pretty sure this could be pretty helpful. :-)

**Comment by fvet** [comment]
I think this can already be achieved with below rules
- https://github.com/StefanMaron/BusinessCentral.LinterCop/wiki/LC0011
- https://github.com/StefanMaron/BusinessCentral.LinterCop/wiki/LC0018
- https://github.com/StefanMaron/BusinessCentral.LinterCop/wiki/LC0025

LC0011 forces the codeunit to carry the 'Access' property and if not set to 'Internal', it would require you to document public procedures (LC0025). If set to internal and events are exposed, LC0018 will show up.

**Reply by someC0d3r** [reply]
Oh okay! Didn't know about that, I think those already existing rules can help out too. Thank you!

I'll close the discussion. :-)
